interop_binaries/tests/end_to_end.rs

```diff
@@ -422,6 +423,9 @@ async fn run(
         }
     };
 
+    // Wait a few seconds to allow the aggregation process to start for all uploaded reports.
+    sleep(StdDuration::from_secs(5)).await;
```

Comment:
I filed https://github.com/divviup/janus/issues/1380 for improvements here. I agree this is far from an ideal solution; I suspect the interop test API is not quite sufficient given the lack of semantics guaranteed by DAP here.

I don't think we need a protocol change; I think this can be handled by implementations. Though this is a gap in specified behavior between the aggregator & collector, I think "reasonable" aggregator/collector implementations will get the correct behavior in practice:
* For fixed-size tasks, there's no problem -- the aggregator is responsible for providing full batches to a `current-batch` query, so the collector can't produce a "too early" collection request.
* For time-interval tasks, if the collector doesn't issue a collection request until after the end of the batch in question, there won't be a problem modulo clock-skewed straggler reports.

Still, we can improve Janus behavior for the case where a collection request is issue too early:
* We could deny these requests entirely. This is an easy choice to implement, but it's the least collector-friendly & we'd still need to update this integration test.
* We could avoid processing these requests (and still allow aggregation into the batch to occur) until after a suitable point in time, most likely the end of the batch or the end of the batch + max clock skew. This is trickier to implement, but would allow collectors to issue collection requests then poll them at any time and get the "right" behavior.

Reply:
(in the long run, we might want to wait until all relevant reports are associated with an aggregation job, too. otherwise, if the aggregation job creator is running behind somehow, even waiting until the end of the batch or any amount of time afterwards might not be enough -- we'd still end up leaving some reports unaggregated. we'd need to come up with a performant way to implement these semantics if we want them.)

Reply:
re: collectors knowing when it's safe to issue a collect request: I see now how I misread the design in #1283 and the code here. I thought that a batch being in state `CLOSING` meant no new aggregation jobs could be scheduled for it, but that's not the case. So the only thing to keep thinking about here is improving the potential race condition which is covered by #1380.